enhancements/microshift/microshift-router-configuration-performance.md

```diff
@@ -0,0 +1,199 @@
+| tuningOptions -> threadCount                 | defines the number of threads created per HAProxy process..                                                                                                                                     | 4       |
+| tuningOptions -> maxConnections              | defines the maximum number of simultaneous connections that can be established per HAProxy process.                                                                                             | 50000   |
+| LogEmptyRequests                             | specifies how connections on which no request is received should be logged.                                                                                                                     | Log     |
+| HTTPEmptyRequestsPolicy                      | indicates how HTTP connections for which no request is received should be handled.                                                                                                              | Respond |
+| HTTP2IsEnabled                               | enables http/2 in ingress controller                                                                                                                                                            | false   |
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+    tuningOptions:
+        headerBufferBytes: 32768
+        headerBufferMaxRewriteBytes: 8192
+        healthCheckInterval: 5000m
+        clientTimeout: 30s
+        clientFinTimeout: 1s
+        serverTimeout: 30s 
+        serverFinTimeout: 1s
+        tunnelTimeout: 1h
+        tlsInspectDelay: 5s
+        threadCount: 4
+        maxConnections: 50000
+    httpCompressionMimeTypes: none
+    LogEmptyRequests: Log
+    forwardedHeaderPolicy: Append
+    HTTPEmptyRequestsPolicy: Respond
+    HTTP2IsEnabled: false
+```
+
+For more information check each individual section.
+
+#### Hypershift / Hosted Control Planes
+N/A
+### Topology Considerations
+N/A
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+
+
+### Risks and Mitigations
+-  Incorrect ingress configuration can cause network  disruption and unpredictable behavior , the documentation should warn about dangers of changing the defaults. 
```

Comment:
At least one warning is listed in OCP docs: https://docs.openshift.com/container-platform/4.17/networking/ingress-operator.html
Should we do the same?